apps/src/netsim/NetSimSendPanel.js

```diff
@@ -110,209 +110,105 @@ NetSimSendPanel.prototype.render = function () {
  */
-NetSimSendPanel.prototype.bindElements_ = function () {
-  var rootDiv = this.getBody();
-
-  var shortNumberFields = [
-    'toAddress',
-    'fromAddress',
-    'packetIndex',
-    'packetCount'
-  ];
-
-  var rowTypes = [
-    {
-      typeName: 'binary',
-      shortNumberAllowedCharacters: /[01]/,
-      shortNumberConversion: binaryToInt,
-      messageAllowedCharacters: /[01\s]/,
-      messageConversion: minifyBinary
-    },
-    {
-      typeName: 'hexadecimal',
-      shortNumberAllowedCharacters: /[0-9a-f]/i,
-      shortNumberConversion: hexToInt,
-      messageAllowedCharacters: /[0-9a-f\s]/i,
-      messageConversion: hexToBinary
-    },
-    {
-      typeName: 'decimal',
-      shortNumberAllowedCharacters: /[0-9]/,
-      shortNumberConversion: parseInt,
-      messageAllowedCharacters: /[0-9\s]/,
-      messageConversion: function (decimalString) {
-        return decimalToBinary(decimalString, this.currentChunkSize_);
-      }.bind(this)
-    },
-    {
-      typeName: 'ascii',
-      shortNumberAllowedCharacters: /[0-9]/,
-      shortNumberConversion: parseInt,
-      messageAllowedCharacters: /./,
-      messageConversion: function (asciiString) {
-        return asciiToBinary(asciiString, this.currentChunkSize_);
-      }.bind(this)
-    }
-  ];
-
-  rowTypes.forEach(function (rowType) {
-    var tr = rootDiv.find('tr.' + rowType.typeName);
-    var rowUIKey = rowType.typeName + 'UI';
-    this[rowUIKey] = {};
-    var rowFields = this[rowUIKey];
-
-    // We attach focus (sometimes) to clear the field watermark, if present
-    // We attach keypress to block certain characters
-    // We attach keyup to live-update the widget as the user types
-    // We attach blur to reformat the edited field when the user leaves it,
-    //    and to catch non-keyup cases like copy/paste.
-
-    shortNumberFields.forEach(function (fieldName) {
-      rowFields[fieldName] = tr.find('input.' + fieldName);
-      rowFields[fieldName].keypress(
-          whitelistCharacters(rowType.shortNumberAllowedCharacters));
-      rowFields[fieldName].keyup(
-          this.makeKeyupHandler(fieldName, rowType.shortNumberConversion));
-      rowFields[fieldName].blur(
-          this.makeBlurHandler(fieldName, rowType.shortNumberConversion));
-    }, this);
-
-    rowFields.message = tr.find('textarea.message');
-    rowFields.message.focus(removeWatermark);
-    rowFields.message.keypress(
-        whitelistCharacters(rowType.messageAllowedCharacters));
-    rowFields.message.keyup(
-        this.makeKeyupHandler('message', rowType.messageConversion));
-    rowFields.message.blur(
-        this.makeBlurHandler('message', rowType.messageConversion));
-  }, this);
-
-  this.bitCounter = rootDiv.find('.bit-counter');
-
-  this.sendButton_ = rootDiv.find('#send_button');
-  this.sendButton_.click(this.onSendButtonPress_.bind(this));
+NetSimSendPanel.prototype.resetPackets_ = function () {
+  this.packetsDiv_.empty();
+  this.packets_ = [];
```

Comment:
If so, I think we may use the idiom `this.packets_.length = 0` elsewhere (where goog.array.clear is not available). If not, all good